pages/DSNP/Types/Update.md

```diff
@@ -0,0 +1,45 @@
+# Update Announcement
+
+An Update Announcement is a way to note intent to update previously
+announced content.
+
+## Fields
+
+| Field | Description | Data Type | Serialization | Parquet Type | Bloom Filter |
+| ----- | ----------- | --------- | ------------- | ------------ | ------------ |
+| announcementType | Announcement Type Enum (`6`) | enum | [decimal](../Serializations.md#decimal) | `INT32` | no |
+| createdAt | milliseconds since Unix epoch | 64 bit unsigned integer | [decimal](../Serializations.md#decimal) | `UINT_64` | no
+| fromId | id of the user creating the announcement | 64 bit unsigned integer | [decimal](../Serializations.md#decimal) | `UINT_64` | YES
+| contentHash | keccak-256 hash of updated content | 32 bytes | [hexadecimal](../Serializations.md#hexadecimal) | `BYTE_ARRAY` | YES
+| url | updated content URL | UTF-8 | [UTF-8](https://datatracker.ietf.org/doc/html/rfc3629) | `UTF8` | no
+| signature | creator signature | 65 bytes | [hexadecimal](../Serializations.md#hexadecimal) | `BYTE_ARRAY` | no
```

Comment:
so this is missing one of the most important fields. Which announcement or content is getting updated?

Reply:
So I think we could go one of two ways on this.

One is to have an identifier that points to a previous announcement. However, if that previous announcement is itself an Update, the client will have to look up (potentially multiple) other announcements. This is possible, but it could be expensive. Still possible, though.

Another option is to leave out the pointer altogether. We simply broadcast intent to update with a URL that points to the most recent content. Clients, after receiving an Update announcement, could simply update the URL / content. I decided to go with this approach here bc it seemed simpler.

Thoughts?

Reply:
Do client needs to know edit history ? Will there be a use case for that, in that case we may link to previous one and follow the trail

Reply:
Please see updates.